test/test_check_mcp.bats

```diff
@@ -0,0 +1,32 @@
+setup() {
+    load 'lib/bats-support/load'
+    load 'lib/bats-assert/load'
+    load 'lib/common.bash'
+}
+
+@test "Check MCP status" {
+
+    # Define the pools to check
+    pools=("master" "worker")
+
+    # Track failures
+    failed=0
+
+    for pool in "${pools[@]}"; do
+        echo "Checking MCP status for: $pool" >&3
+        upgraded_status=$(${KUBECTL} get mcp "$pool" --no-headers | awk '{print $3}')
```

Comment:
By asking for the field explicitly your command is more robust (e.g., it will continue to work even if the output of `kubectl get mcp` changes):

```suggestion
upgraded_status=$(${KUBECTL} get mcp "$pool" -o jsonpath='{.status.conditions[?(.type=="Updated")].status}')
```